sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1663,8 +1663,12 @@ object ReplaceDeduplicateWithAggregate extends Rule[LogicalPlan] {
 /**
  * Replaces logical [[Intersect]] operator with a left-semi [[Join]] operator.
  * {{{
- *   SELECT a1, a2 FROM Tab1 INTERSECT SELECT b1, b2 FROM Tab2
- *   ==>  SELECT DISTINCT a1, a2 FROM Tab1 LEFT SEMI JOIN Tab2 ON a1<=>b1 AND a2<=>b2
```

Comment:
This seems like we push down the aggregate operator through the join operator. It's usually beneficial, but can cause perf regression if the join children do not have many duplications and the aggregate operator can't reduce the data volume.

Reply:
Could we add a config?

Reply:
Another similar case.
Origin SQL:
```sql
SELECT DISTINCT t.at_prod_ref_id, d.week_beg_dt
FROM   table1 t
       JOIN table2 d
         ON d.dt BETWEEN t.start_dt AND t.end_dt
WHERE  d.dt BETWEEN '2019-01-01' AND CURRENT_DATE()
       AND cid IN ( 0, 3, 15, 77, 71, 101, 186, 100 ); 
```
Push down distinct through `BroadcastNestedLoopJoin`:
```sql
SELECT DISTINCT t.at_prod_ref_id, d.week_beg_dt FROM
(  SELECT DISTINCT at_prod_ref_id, start_dt, end_dt
   FROM table1 WHERE cid IN ( 0, 3, 15, 77, 71, 101, 186, 100 )
) t
JOIN (
  SELECT DISTINCT dt, week_beg_dt FROM table2 WHERE dt BETWEEN '2019-01-01' AND current_date()
) d ON d.dt BETWEEN t.start_dt AND t.end_dt;
```

![image](https://user-images.githubusercontent.com/5399861/110332622-882d9500-805b-11eb-9b6f-31f17b109fdb.png)

Reply:
We cannot use column stats (`distinctCount`) to determine whether the optimizer push down it or not?

Reply:
Do you mean only push down distinct when CBO is enabled and all column stats are exist and  there are duplicate values?

Reply:
Yea, I meant so, `spark.sql.cbo.enabled` or `spark.sql.cbo.planStats.enabled`

Reply:
@cloud-fan What do you think? As far as I know, there is no column stats in our production. I don't know the status of other companies.